docs/source/daml2ts/index.rst

```diff
@@ -0,0 +1,291 @@
+        decoder: () => jtv.object({
+          name: daml.Text.decoder(),
+          party: daml.Party.decoder(),
+          aage: daml.Int.decoder(),
+        }),
+      })
+
+Generic
+*******
+
+In the above case, ``Person`` is a monomorphic type. If we change it slightly so that it is polymorphic in the type of ``age``, it would look like this.
+
+.. code-block:: daml
+   :linenos:
+
+   data Person a =
+     Person
+       name : Text
+       party : Party
+       age : a
+
+In this case, the generated Typescript would read like the following.
+
+.. code-block:: typescript
+   :linenos:
+
+   export type Person<a> = {
+     name: string;
+     party: daml.Party;
+     age: a;
+   }
+
+   export const Person = <a>(a: daml.Serializable<a>): daml.Serializable<Person<a>> => ({
+     decoder: () => jtv.object({
+       name: daml.Text.decoder(),
+       party: daml.Party.decoder(),
+       age: a.decoder(),
+     }),
+   })
+
+Tuples
+~~~~~~
+
+Suppose the following DAML definition.
+
+.. code-block:: daml
+   :linenos:
+
+   data Pair a b = Pair (a, b)
+
+In Typescript, the above definition renders as follows.
+
+.. code-block:: typescript
+   :linenos:
+
+   import * as pkg40f452260bef3f29dede136108fc08a88d5a5250310281067087da6f0baddff7_DA_Types from './../40f452260bef3f29dede136108fc08a88d5a5250310281067087da6f0baddff7/DA/Types';
+
+   export type Pair<a, b> =
+     |  { tag: 'Pair'; value: pkg40f452260bef3f29dede136108fc08a88d5a5250310281067087da6f0baddff7_DA_Types.Tuple2<a, b> }
+
+   export const Pair = <a, b>(a: daml.Serializable<a>, b: daml.Serializable<b>): daml.Serializable<Pair<a, b>> => ({
+     decoder: () => jtv.oneOf<Pair<a, b>>(
+       jtv.object({tag: jtv.constant('Pair'), value: jtv.lazy(() => pkg40f452260bef3f29dede136108fc08a88d5a5250310281067087da6f0baddff7_DA_Types.Tuple2(a, b).decoder())}),
+     ),
+   });
+
+What we are interested in here is the treatment of the tuple type ``(a, b)``. As the above shows, its presence has been captured by a type ``Tuple2<a, b>`` from a package generated from the DAML standard library. Here is the definition.
+
+.. code-block:: typescript
+   :linenos:
+
+   export type Tuple2<t1, t2> = {
+     _1: t1;
+     _2: t2;
+   }
+   export const Tuple2 = <t1, t2>(t1: daml.Serializable<t1>, t2: daml.Serializable<t2>): daml.Serializable<Tuple2<t1, t2>> => ({
+     decoder: () => jtv.object({
+       _1: t1.decoder(),
+       _2: t2.decoder(),
+     }),
+   })
+
+The DAML type that the above is generated from is exactly this.
+
+.. code-block:: daml
+   :linenos:
+
+   data Tuple2 a b =
+       Tuple2 with
+         _1 : a
+         _2 : b
+
+.. note::
+   The general rule is that DAML tuples are represented in Typescript as simple objects with fields labeled ``_1``, ``_2`` and so on.
+
+The DAML type ``Pair a b`` has been modeled in Typescript as a discriminated union with a single case. We'll see more discriminated unions in the next section.
```

Comment:
You call them `discriminated union`s here only to start with a section called `Variants` below. I think it would make sense to be consistent.

Reply:
Ha. The mismatch is that TS terminology is discriminated union types and the sections are named for DAML types.  I'll work something out 😃